docs/contributing/setup-development-machine-with-vagrant.md

```diff
@@ -4,325 +4,109 @@
 
-```ruby
-Vagrant.configure("2") do |config|
-  # config.vm.box = "ubuntu/focal64"     # Ubuntu 20.04 Focal Fossa (non CO-RE)
-  # config.vm.box = "ubuntu/hirsute64"   # Ubuntu 21.04 Hirsute Hippo (CO-RE)
-  # config.vm.box = "ubuntu/impish64"    # Ubuntu 21.10 Impish Indri (CO-RE)
-  config.vm.box = "ubuntu/jammy64"       # Ubuntu 22.04 Jammy Jellyfish (CO-RE)
-...
-```
+  ```bash
+  export VM_TYPE=dev
+  ```
 
-Sometimes you may want to test Tracee with a non CO-RE distribution. You can do
-that by editing the Vagrantfile and modifying the `config.vm.box` property. For
-example, you can switch to Ubuntu Linux 20.04 Focal Fossa as follows:
-
-```ruby
-Vagrant.configure("2") do |config|
-  config.vm.box = "ubuntu/focal64"       # Ubuntu 20.04 Focal Fossa (non CO-RE)
-  # config.vm.box = "ubuntu/hirsute64"   # Ubuntu 21.04 Hirsute Hippo (CO-RE)
-  # config.vm.box = "ubuntu/impish64"    # Ubuntu 21.10 Impish Indri (CO-RE)
-  # config.vm.box = "ubuntu/jammy64"     # Ubuntu 22.04 Jammy Jellyfish (CO-RE)
-...
-```
-
-This change requires re-provisioning the development machine:
+- **Testing Environment:**
 
-```console
-vagrant destroy
-vagrant up
-```
+  ```bash
+  export VM_TYPE=test
+  ```
 
-!!! Attention
-    Ubuntu Focal distribution has introduced BTF information to their recent
-    kernels, allowing eBPF CO-RE capable code to run. If you're willing to test
-    non CO-RE kernels, make sure to use an older kernel that does not provide
-    the `/sys/kernel/btf/vmlinux` file.
+## Configure Resource Allocation (Optional)
 
-## Deploy Tracee with Postee on Kubernetes
+Customize the VM's resources by setting the following environment variables:
 
-The development machine described by Vagrantfile pre-installs [MicroK8s] Kubernetes cluster, which is suitable for testing Tracee.
+- `VM_PROC`: Number of virtual processors. Defaults to half of the host's processors. Example:
 
-```console
-microk8s status
-```
+  ```bash
+  export VM_PROC=4
+  ```
 
-```text
-microk8s is running
-high-availability: no
-  datastore master nodes: 127.0.0.1:19001
-  datastore standby nodes: none
-...
-```
+- `VM_MEM`: Memory in gigabytes. Defaults to 8GB. Example:
 
-There's also the [kubectl] command installed and configured to communicate with
-the cluster:
+  ```bash
+  export VM_MEM=16
+  ```
 
-```console
-kubectl get nodes -o wide
-```
+## Start the VM
 
-```
-NAME           STATUS   ROLES    AGE   VERSION   INTERNAL-IP   EXTERNAL-IP   OS-IMAGE             KERNEL-VERSION      CONTAINER-RUNTIME
-ubuntu-jammy   Ready    <none>   40m   v1.26.1   10.0.2.15     <none>        Ubuntu 22.04.2 LTS   5.15.0-69-generic   containerd://1.6.8
-```
+Run the following command to start the VM:
 
-Create a new namespace called `tracee-system`:
+  ```bash
+  vagrant up
+  ```
 
-```console
-kubectl create ns tracee-system
-```
+Vagrant will download the base box (Ubuntu 22.04), provision the VM, and install all required dependencies. This process may take some time.
```

Comment:
avoid mentioning versions in this case so it will be easier to bump Vagrant without changing those references.